### ISSUE TITLE
VCF merging strikes again

### ISSUE THREAD

**Issue by cassimons** [populationgenomics/production-pipelines#287]
At the end of 2022 we had several issues with VCF gather jobs. The culmination of this was [this PR ](https://github.com/populationgenomics/production-pipelines/pull/241) where we swapped away from GATK GatherVcfsCloud to using `bcftools merge`. This change resolved the issue we were having with sites-only vcf merging but it seems it was not tested on any full vcfs.

Running the seqr pipeline now the merge VCFs job is throwing this [error](https://batch.hail.populationgenomics.org.au/batches/421182/jobs/1):
```
Error: Duplicate sample names (CPG11312), use --force-samples to proceed anyway.
```

I confirmed the part VCFs do not contain duplicate samples. I then looked at the [README](https://samtools.github.io/bcftools/bcftools.html#merge) for `bcftools merge`

```
Merge multiple VCF/BCF files from non-overlapping sample sets to create one multi-sample file. For example, when merging file A.vcf.gz containing samples S1, S2 and S3 and file B.vcf.gz containing samples S3 and S4, the output file will contain five samples named S1, S2, S3, 2:S3 and S4.

Note that it is responsibility of the user to ensure that the sample names are unique across all files.
```
Basically, this tool is designed to merge vcfs tools sharded by sample NOT sharded by interval. This worked fine for sites-only vcfs only because they did not have any samples to conflict.

Options:
 - Revert to GatherVcfsCloud. This should fix this error, but will not address the previous issue we have had of variants spanning interval boundaries resulting in unsorted vcf output.
 - Swap to use `bcftools concat -a`. I suspect this was the tool Vlad intended to use. [It is designed for this job](https://samtools.github.io/bcftools/bcftools.html#concat) and can handle the variants spanning interval boundry issue above.
